src/include/migraphx/run_loop.hpp

```diff
@@ -0,0 +1,124 @@
+#ifndef MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+#define MIGRAPHX_GUARD_RTGLIB_RUN_LOOP_HPP
+
+#include <string>
+#include <migraphx/instruction_ref.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/argument.hpp>
+#include <migraphx/context.hpp>
+#include <migraphx/module.hpp>
+#include <migraphx/config.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+template <class LoopModel, class T>
+argument run_loop(const LoopModel& model,
+                  T& ctx,
+                  std::vector<argument> args,
+                  const std::vector<module_ref>& mods,
+                  const std::function<std::vector<argument>(
+                      module_ref&, const std::unordered_map<std::string, argument>&)>& run)
+{
+    auto get_output_index = [](const std::string& name) {
+        std::string out_prefix = "#output_";
```

Comment:
The naming of output indices as `#output_` is specific to the GPU backend. Other backends could name them differently. Why do you need to check for the output params? Maybe this should added to the loop_model, maybe a function like `std::map<std::string, int> get_output_params(const module& m)`?

Reply:
moved this to the gpu implementation